rocAL/rocAL_pybind/amd/rocAL/fn.py

```diff
@@ -0,0 +1,547 @@
+    if crop_shape is None:
+        crop_shape = []
+    else:
+        crop_shape = crop_shape
+    if input_shape is None:
+        input_shape = []
+    else:
+        input_shape = input_shape
+    scaling = scaling if scaling else [1.0, 1.0]
+    thresholds = thresholds if thresholds else [0.0]
+    crop_begin = []
+    if(len(crop_shape) == 0):
+        has_shape = False
+        crop_width = 0
+        crop_height = 0
+    else:
+        has_shape = True
+        crop_width = crop_shape[0]
+        crop_height = crop_shape[1]
+    scaling = b.CreateFloatUniformRand(scaling[0], scaling[1])
+    aspect_ratio = b.CreateFloatUniformRand(aspect_ratio[0], aspect_ratio[1])
+
+    # pybind call arguments
+    kwargs_pybind = {"all_boxes_above_threshold":all_boxes_above_threshold, "no_crop": allow_no_crop, "p_aspect_ratio":aspect_ratio, "has_shape":has_shape, "crop_width":crop_width, "crop_height":crop_height, "num_attemps":num_attempts, "p_scaling":scaling, "total_num_attempts":total_num_attempts }
+    random_bbox_crop = b.RandomBBoxCrop(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+
+    return (random_bbox_crop,[],[],[])
+
+def bb_flip(*inputs, bytes_per_sample_hint = 0, horizontal = 1, ltrb = False, preserve =  False,seed = -1, vertical = 0, device = None):
+    #Dummy Node
+    return []
+#TODO: To be Handled Later
+def one_hot(*inputs, bytes_per_sample_hint=0, dtype=types.FLOAT, num_classes=0, off_value=0.0,
+            on_value=1.0, preserve=False, seed=-1,  device=None):
+    Pipeline._current_pipeline._numOfClasses = num_classes
+    Pipeline._current_pipeline._oneHotEncoding = True
+    return ([])
+
+def box_encoder(*inputs, anchors, bytes_per_sample_hint=0, criteria=0.5, means=None, offset=False, preserve=False, scale=1.0, seed=-1, stds=None ,device = None):
+    means = means if means else [0.0, 0.0, 0.0, 0.0]
+    stds = stds if stds else [1.0, 1.0, 1.0, 1.0]
+    kwargs_pybind ={"anchors":anchors, "criteria":criteria, "means":means, "stds":stds, "offset":offset, "scale":scale}
+    box_encoder = b.BoxEncoder(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+    Pipeline._current_pipeline._BoxEncoder = True
+    return (box_encoder , [])
+
+def color_temp(*inputs, adjustment_value=50, device=None, preserve = False):
+    # pybind call arguments
+    adjustment_value = b.CreateIntParameter(adjustment_value) if isinstance(adjustment_value, int) else adjustment_value
+    kwargs_pybind = {"input_image0": inputs[0],"is_output": False, "adjustment_value": adjustment_value}
+    color_temp_transformed_image = b.ColorTemp(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+    return (color_temp_transformed_image)
+
+def nop(*inputs, device=None, preserve = False):
+    # pybind call arguments
+    kwargs_pybind = {"input_image0": inputs[0],"is_output": False }
+    nop_trandsformed_image = b.raliNop(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
```

Comment:
better to name as nop_output (instead of other name). Applicable to other fns in the same file

Reply:
please also check spelling for "transformed" if retaining name

Reply:
Done